sbt_common/messaging/src/test/scala/uk/ac/wellcome/messaging/message/MessageWriterTest.scala

```diff
@@ -93,4 +95,27 @@ class MessageWriterTest
     }
   }
 
+  it("gives distinct s3 keys when sending the same message twice") {
+    withLocalSnsTopic { topic =>
+      withLocalS3Bucket { bucket =>
+        withExampleObjectMessageWriter(bucket, topic) { messageWriter =>
+          val eventualAttempt1 = messageWriter.write(message, subject)
+          Thread.sleep(2)
```

Comment:
I guess the `sleep` here is to make sure these occur in different epochs and hence different keys?  Perhaps make this a bit clearer?

Reply:
👍